### ISSUE TITLE
PHP Reflect.getProperty on method throws error

### ISSUE THREAD

**Issue by andyli** [HaxeFoundation/haxe#4915]
```haxe
class Main {
	function new():Void {}

	function method():Void {}

	var field = "field";

	var prop(get, null):String;
	function get_prop() return "prop";

	static function main() {
		var m = new Main();
		// Reflect.getProperty(m, "field"); // no problem
		// Reflect.getProperty(m, "prop");  // no problem

		Reflect.getProperty(m, "method");   // error
	}
}
```
Error:
```
uncaught exception: Undefined property: Main::$method (errno: 8) in /Users/andy/Documents/workspace/erazor/bin/lib/Reflect.class.php at line #50Undefined property: Main::$method

Called from Reflect::getProperty
Called from Main::main
[Finished in 0.2s]
```

This issue exists in both haxe dev and haxe 3.2.1, and is the cause of the hscript issue at https://github.com/HaxeFoundation/hscript/commit/98184bf9bf616fb0cc7ee5759f15ff11f77809bb#commitcomment-15784997

P.S. I found that the Reflect unit test is [disabled for PHP](https://github.com/HaxeFoundation/haxe/blob/development/tests/unit/src/unitstd/Reflect.unit.hx#L1)... Any reason?

**Comment by Simn** [comment]
I don't know what to do with the PHP target right now, there's nobody who can address issues like this without breaking something else.

**Comment by mockey** [comment]
I think I have a simple fix for this issue at least.
I tried to enable Reflect.unit for php and I see t fails at other issues (`isObject`, `isEnumValue`). Should these be fixed as well?

Also there seem to be some tests for `getProperty` but they are commented out:
https://github.com/HaxeFoundation/haxe/blob/development/tests/unit/src/unitstd/Reflect.unit.hx#L39
https://github.com/HaxeFoundation/haxe/blob/development/tests/unit/src/unitstd/Reflect.unit.hx#L54
Were there problems with other targets?

**Comment by Simn** [comment]
Looks like these have been commented out from the start: https://github.com/HaxeFoundation/haxe/commit/833cb2b3dd2bc404a6f425eed216287536cc1257

Unfortunately I don't remember why, specifying Reflect in particular was a royal pain and spawned several issues.

**Comment by mockey** [comment]
So, uncomment them and see what happens? :-)

Also `isObject` is really supposed to be true for an Enum and a Class?
https://github.com/HaxeFoundation/haxe/blob/development/tests/unit/src/unitstd/Reflect.unit.hx#L124
https://github.com/HaxeFoundation/haxe/blob/development/tests/unit/src/unitstd/Reflect.unit.hx#L125

**Comment by Simn** [comment]
I think so because you can access its statics like that and also use it as subject to `Refiect.field` or something.

**Comment by mockey** [comment]
So, `isObject` and `isEnumValue` never worked for PHP?

**Comment by Simn** [comment]
I don't know, I just work here...

**Comment by mockey** [comment]
OK. Reflect.unit passes now with PHP.
I removed the `if !php` and the comments in these lines:
https://github.com/HaxeFoundation/haxe/blob/development/tests/unit/src/unitstd/Reflect.unit.hx#L39
https://github.com/HaxeFoundation/haxe/blob/development/tests/unit/src/unitstd/Reflect.unit.hx#L54
https://github.com/HaxeFoundation/haxe/blob/development/tests/unit/src/unitstd/Reflect.unit.hx#L55
Do you want a PR like this or do you want to keep the comments?

**Comment by Simn** [comment]
If it works then go for it! Our CI will tell us if there are any issues on other targets.

**Comment by andyli** [comment]
Fixed by #4921.
The test case of this issue is covered in the unit test ([Reflect.unit.hx](https://github.com/HaxeFoundation/haxe/blob/development/tests/unit/src/unitstd/Reflect.unit.hx#L38)):
```haxe
Reflect.getProperty(c, "func")() == "foo";
```
